scipy/stats/_stats_py.py

```diff
@@ -540,41 +540,39 @@ def _mask_to_limits(a, limits, inclusive):
     limits : (float or None, float or None)
         A tuple consisting of the (lower limit, upper limit).  Values in the
         input array less than the lower limit or greater than the upper limit
-        will be masked out. None implies no limit.
+        will be replaced with `np.nan`. None implies no limit.
     inclusive : (bool, bool)
         A tuple consisting of the (lower flag, upper flag).  These flags
         determine whether values exactly equal to lower or upper are allowed.
 
-    Returns
-    -------
-    A MaskedArray.
-
-    Raises
-    ------
-    A ValueError if there are no values within the given limits.
-
     """
+    if limits is None:
+        return a
+    a = a.copy()
```

Comment:
Since you're alreay copying, might as well ensure that the dtype is inexact so you won't have to copy again.

Please check me here, but something like:
```suggestion
    a = a.copy() if np.issubdtype(a, np.inexact) else a.astype(np.float64)
```

Reply:
Actually, I just realized that the functions preserved integer dtypes when no NaNs are required. For example,

```python
from stats import tmin

tmin([1,2,3], lowerlimit=2)  # outputs 2 of type `np.int_` on branch main
```

This change (and the changes I have made currently) wouldn't preserve that. I guess one of the reasons to use masking here was this. Not sure if I should re-introduce it or just cast the data type back to the input dtype before returning. Let me know what you think here. I guess the latter option doesn't sound too bad since we don't allow non-numerical data types.

Reply:
Good catch, but let's just cast it back.